worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/DefaultBlockParser.java

```diff
@@ -206,7 +207,36 @@ private static Map<Property<?>, Object> parseProperties(BlockType type, String[]
     public Stream<String> getSuggestions(String input, ParserContext context) {
         final int idx = input.lastIndexOf('[');
         if (idx < 0) {
-            return SuggestionHelper.getNamespacedRegistrySuggestions(BlockType.REGISTRY, input);
+            ArrayList<String> playerSpecificBlocks = new ArrayList<>();
+
+            var actor = context.getActor();
+            if (actor instanceof Player player) {
+                try {
```

Comment:
I'm a little confused by these try statements wrapping it, and why you're calling the getBlockInHand methods here? if it's a player, you should be able to _always_ add `hand` and `offhand` to this list.

For `pos1`, it's realistically better to just always provide it as well. It's not only players who can have selections, so pos1 is _always_ a valid selection. It's better to suggest it and have it give the "need to select a region" message upon running it, as that informs the user that it's actually possible.